.github/workflows/main.yaml

```diff
@@ -349,3 +350,42 @@ jobs:
         run: |
           export DOCKER_BUILD_AND_SYSTEM_PRUNE="true"
           make image 
+  
+  container_runtime_test:
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        container-runtime: ['cri-o', 'docker']
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install and configure container runtime
+        run: |
+          if [ "${{ matrix.container-runtime }}" == "docker" ]; then
+            sudo apt-get update
+            sudo apt-get install \
+                apt-transport-https \
+                ca-certificates \
+                curl \
+                gnupg \
+                lsb-release
+            curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+            echo \
+              "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+              $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+            sudo apt-get update
+            sudo apt-get install docker-ce docker-ce-cli containerd.io cri-dockerd
+          elif [ "${{ matrix.container-runtime }}" == "cri-o" ]; then
+            sudo apt update
+            sudo apt install -y cri-o cri-o-runc
+            sudo systemctl start crio
+            sudo systemctl enable crio
+          fi
+
+      - name: Run tests
+        run: make e2e  
```

Comment:
I believe there are several steps that need to be performed before executing "make e2e". We can refer to lines 162-238 in the e2e test for guidance. These steps may include `installing dependencies`, `retrieving the saved kubeedge/build-tools image`, l`oading the kubeedge/build-tools image using Docker`, and so on. Additionally, when executing "make e2e", the following environment variables also need to be set.
![image](https://github.com/kubeedge/kubeedge/assets/48788150/f19b7f50-9912-468b-8441-233398091583)